200: Added AI_CONTRIBUTIONS.md policy for AI-generated contributions

FreezinGaits opened the pull request:
## Description of Changes

This pull request adds `AI_CONTRIBUTIONS.md` to establish a policy for managing AI-generated contributions to the DISCOVER-Cookbook, addressing Issue #200. The policy ensures that AI-assisted content aligns with the project’s mission to foster collaborative, high-quality, and inclusive content for diversity and inclusion (D&I) in scientific computing. Key features include:

- **Transparency**: Requires contributors to disclose AI tool usage (e.g., ChatGPT, GitHub Copilot) in commit messages and PR descriptions using `Generated-By` or `Assisted-By` tags.
- **Human Oversight**: Mandates human review and validation of AI-generated content to ensure quality and relevance.
- **Permitted Uses**: Allows AI for drafting content, code, translations, and research, with human verification.
- **Prohibited Practices**: Prevents unattributed AI content, low-quality submissions, or violations of open-source licensing and NumFOCUS’s Code of Conduct.
- **Review and Enforcement**: Outlines an enhanced review process and consequences for policy violations, supporting community-driven standards.

The policy was informed by research into AI contribution guidelines from Matplotlib, Apache Software Foundation, Linux Foundation, OpenInfra Foundation, Elsevier, and ACM, ensuring alignment with open-source best practices. It supports the DISCOVER-Cookbook’s goals of fostering authentic community contributions and maintaining ethical standards.

This contribution is part of my Google Summer of Code (GSoC) project, addressing the need for a clear policy on AI-generated content as requested in Issue #200.

## Related Issue

Fixes #200

## Type of Change

- [x] 🛠️ Other — Adds a new policy document (`AI_CONTRIBUTIONS.md`) to address AI-generated contributions, as requested in Issue #200.

## Additional Notes (Optional)

- The policy is designed as a living document, with provisions for regular updates based on community feedback, technological advancements, and the DISC Committee’s guidance.
- I acknowledge that the policy could be further tailored to explicitly address the versioned content model, crowd-sourced translations, and 3rd edition editorial process (Issue #199). I’m happy to revise based on maintainer feedback to ensure closer alignment with these initiatives.
- A research summary detailing findings from Matplotlib, Elsevier, ACM, and other sources is available upon request for GSoC evaluation.
- I propose discussing the policy at the DISC unconference to gather community input, aligning with the project’s editorial process.
- Feedback from @aterrel, @melissawm, and the DISC Committee is welcome to refine the policy further.

## Checklist

- [x] I have searched [open pull requests](https://github.com/numfocus/DISCOVER-Cookbook/pulls) to avoid duplicates.
- [x] I have built the book locally using the [build instructions](https://github.com/numfocus/DISCOVER-Cookbook#how-to-run-the-book-locally) and verified my changes work as expected.
- [x] I have written clear, conventional commit messages, as per the [Pull Request Guidelines](https://github.com/numfocus/DISCOVER-Cookbook/blob/main/CONTRIBUTING.md#pull-request-guidelines).

FreezinGaits commented:
Hi @aterrel, @melissawm,

I’m working on Issue #200 for my Google Summer of Code (GSoC) project. I’ve drafted a policy in `AI_CONTRIBUTIONS.md` to address AI-generated contributions, based on research into Matplotlib’s guidelines, Elsevier’s AI policy, and other open-source practices. The policy emphasizes:
- **Disclosure**: Requiring contributors to disclose AI tool usage (e.g., ChatGPT, GitHub Copilot) in PRs and commits.
- **Human Oversight**: Mandating human review to ensure quality and alignment with the DISCOVER-Cookbook’s standards.
- **Alignment with Governance**: Supporting the versioned content model, crowd-sourced translations, and 3rd edition editorial process (Issue #199).
- **Ethical Standards**: Ensuring compliance with NumFOCUS’s Code of Conduct and open-source licensing.

Can you confirm if this approach aligns with the DISC Committee’s vision? Are there specific aspects of the governance model (Issue #199) or project initiatives (e.g., translations, editorial process) I should prioritize? I plan to submit a PR soon and would appreciate any feedback to refine the policy.

Thanks,
FreezinGaits

aterrel commented:
Hi @FreezinGaits,

This does look interesting and much better than what other folks have submitted.

What is the GSOC project? Would love to see the proposal and how this aligns with other ecosystem policies.

FreezinGaits commented:
Hi @aterrel,
Thank you for the feedback! I'm happy to hear the policy draft looks promising.
For my Google Summer of Code (GSoC) project, I'm contributing to NumFOCUS by addressing Issues
The `AI_CONTRIBUTIONS.md` policy draws on research from Matplotlib, Elsevier, ACM, Apache Software Foundation, Linux Foundation, and OpenInfra Foundation to align with open-source best practices. I can share a detailed research summary comparing these ecosystem policies if helpful—please let me know how you'd prefer to receive it (e.g., as a document in the PR or via email).
I’m finalizing the PR with `AI_CONTRIBUTIONS.md` and would appreciate any specific guidance on further aligning with the DISC Committee’s vision or other NumFOCUS project policies.

aterrel commented:
Hmm I don't see such a project officially listed here:
https://summerofcode.withgoogle.com/programs/2025/organizations/numfocus

FreezinGaits commented:
That's right, I apologize for any confusion. I’m working on these issues as proactive contributions, preparing for a potential future GSoC cycle with NumFOCUS. NumFOCUS has been a consistent GSoC participant since 2016, supporting projects like MDAnalysis and Mesa, so I’m hopeful it’ll be part of the next GSoC program.

aterrel reviewed (changes requested):
Needs to add references to the sources for this document. In the note you say you used "Matplotlib, Apache Software Foundation, Linux Foundation, OpenInfra Foundation, Elsevier, and ACM, ensuring alignment with open-source best practices." Add this statment to the end for sure.

danielskatz commented on AI_CONTRIBUTIONS_POLICY.md:
```diff
@@ -0,0 +1,159 @@
+- 🏗️ Formatting and structural help
+
+### 👩💻 Code & Technical Content
+- AI-generated code that is tested and validated
```
We don't really have AI-generated code here, so I think this bullet can be removed

aterrel commented on AI_CONTRIBUTIONS_POLICY.md:
```diff
@@ -0,0 +1,159 @@
+- 🏗️ Formatting and structural help
+
+### 👩💻 Code & Technical Content
+- AI-generated code that is tested and validated
```
AI code needs be referenced. The license issues around AI code is not super clear and we want to know which commits are using it.

aterrel replied:
oh nevermind, I see you show this later, suggested "tested, validated, and referenced"

danielskatz commented on AI_CONTRIBUTIONS_POLICY.md:
```diff
@@ -0,0 +1,159 @@
+
+## 1️⃣1️⃣ Acknowledgments
+
+This policy draws on best practices from:
```
Please add references (links) for each of the following

rivaquiroga commented on AI_CONTRIBUTIONS_POLICY.md:
```diff
@@ -0,0 +1,159 @@
+# 🤖 AI Generated Contributions Policy for DISCOVER Cookbook
```
(e.g., numbering the sections, instead of using an emoji with the number)
